### ISSUE TITLE
Voraussetzungen besser auf Architekten zuschneiden, sie sind eben _keine_ KI/ML-Spezialisten

### ISSUE THREAD

**Issue by mattes3** [isaqb-org/curriculum-swarc4ai#9]
In den Voraussetzungen für die Teilnehmer:innen steht:

> Teilnehmer:innen **sollten** folgende Kenntnisse und/oder Erfahrung mitbringen:
> 
> * Grundlegendes Wissen zu KI, Machine Learning und Data Science
> * Machine Learning Methoden wie Supervised, Unsupervised Learning und Reinforcement Learning, z.B Klassifikation, Clustering, Regression, Dimensionality Reduction
> * Grundsätzliches Wissen über die typischen Algorithmen wie: Linear/logistic regression, decision trees, random forests, support vector machines, K-means clustering, neural networks and deep learning
> * Wissen über Model evaluation: Cross-validation, Metrics (accuracy, precision, recall, F1-score), Bias-variance tradeoff
> * Erfahrung mit typischen Frameworks wie sk-learn, TensorFlow, PyTorch
> * Erfahrung mit Jupyter Notebooks oder ähnlichen Rapid Application Development (RAD) Werkzeugen
> * Wissen über die technischen Verfahren hinter KI, einschließlich: Maschinelles Lernen inkl. Deep Learning und LLMs, NLP-Ansätze (Natural Language Processing), Transformationsmodelle
> * Praktische Erfahrung mit Trainieren von Künstlichen Intelligenz Modellen

Ich finde, der Kurs richtet sich eher an Architekt:innen, nicht an ML-Expert:innen, oder? Bei ersteren sollte "Grundwissen plus ein wenig damit gespielt haben" ausreichen. Im Alltag arbeiten sie ja mit Data Scientists, KI-Expert:innen und Softwareentwickler:innen zusammen, die sind dann für die harten Sachen zuständig und brauchen Erfahrung, das ist klar.

Folgende Voraussetzung sollte man meiner Meinung nach herausnehmen oder als optional kennzeichnen:

* Praktische Erfahrung mit Trainieren von Künstlichen Intelligenz Modellen

**Comment by mahboubagharbi** [comment]
Wird so bleiben. Feedback vom Council.

**Comment by mattes3** [comment]
Hmmm... ist mir nicht klar, warum.

**Comment by mahboubagharbi** [comment]
Eventuell passen wir so an:

•	Ein grundlegendes Verständnis von Künstlicher Intelligenz, Machine Learning und Data Science
•	Vertrautheit mit Machine Learning-Methoden, wie Supervised Learning, Unsupervised Learning und Reinforcement Learning, einschließlich Klassifikation, Clustering, Regression und Dimensionality Reduction
•	Grundkenntnisse über gängige Algorithmen, wie z. B. lineare und logistische Regression, Entscheidungsbäume, Random Forests, Support Vector Machines, K-Means Clustering sowie neuronale Netze und Deep Learning
•	Ein Verständnis der Modellevaluation, einschließlich Cross-Validation, relevanter Metriken (Accuracy, Precision, Recall, F1-Score) und des Bias-Variance Tradeoffs
•	Erfahrung mit typischen Frameworks wie scikit-learn, TensorFlow und PyTorch
•	Vertrautheit mit Jupyter Notebooks oder ähnlichen Werkzeugen zur schnellen Anwendungsentwicklung (Rapid Application Development, RAD)
•	Ein grundlegendes Verständnis der technischen Verfahren hinter Künstlicher Intelligenz, einschließlich Maschinellen Lernens (insbesondere Deep Learning und LLMs) sowie Ansätzen im Bereich Natural Language Processing (NLP) und Transformationsmodellen
•	Praktische Erfahrung im Training von Künstlichen Intelligenz-Modellen

**Comment by magnusse** [comment]
Ich fand Manboubas Vorschlag sehr gut und habe ihn übernommen.

**Comment by programming-wolf** [comment]
Ist das damit erledigt, @mattes3 ?

**Comment by mattes3** [comment]
Die Praktische Erfahrung im Training von Künstlichen Intelligenz-Modellen steht noch als Voraussetzung für Teilnehmer:innen drin. Wollen wir das so lassen? Ich schlage vor, es zu ersetzen durch dieses:

- Praktische Erfahrung in der _Anwendung_ von Künstlichen Intelligenz-Modellen

Das würde dann heißen, dass es ausreicht, wenn jemand Modelle schon angewendet hat, ohne sie selbst trainiert zu haben. M.E. wäre das völlig ausreichend für KI-Architekten (nicht für KI-Entwickler, natürlich!).

**Comment by visenger** [comment]
Vielen Dank für die Anmerkung, @mattes3 !

Aus unserer Sicht, das fundamentale Verständnis wie ein Model trainiert wird, ist essenziell. Data preparation, Feature Engineering, Model Training, Model Evaluation usw. sind Phasen, die in der Architektur eine 1:1 Abbildung haben. 

Teilnehmer:innen, die, diese Voraussetzung nicht mitbringen, müssen mit einer klaren Wissenslücke beim Training rechnen.

**Comment by youngbrioche** [comment]
Hi zusammen, ich finde die Voraussetzungen für die Teilnehmer:innen in der aktuellen Version immer noch zu hart und nicht gut passend für den adressierbaren, relevanten Markt für dieses Training. Die allerwenigsten angehenden Softwarearchitekt:innen werden das erfüllen können. Zudem mischen sich viele Anforderungen mit eigentlichen Lernzielen. Sprich: ich muss schon vorher etwas beherrschen, das ich im Training lernen soll. Ich möchte gerne ein paar Vorschläge für alternative Formulierungen geben. Was denkt Ihr?

> Vertrautheit mit Machine Learning-Methoden, wie Supervised Learning, Unsupervised Learning und
> Reinforcement Learning, einschließlich Klassifikation, Clustering, Regression und Dimensionality
> Reduction

Was meint hier Vertrautheit? Ich würde folgendes vorschlagen:

> Überblickswissen zu Machine Learning-Methoden wie Supervised Learning, Unsupervised Learning und
> Reinforcement Learning.

Hier finde ich die Voraussetzungen auch zu tief formuliert:

> Grundkenntnisse über gängige Algorithmen, wie z. B. lineare und logistische Regression,
> Entscheidungsbäume, Random Forests, Support Vector Machines, K-Means Clustering sowie
> neuronale Netze und Deep Learning

Vorschlag:

> Überblickswissen über gängige Algorithmen, wie z. B. lineare und logistische Regression,
> Entscheidungsbäume, Random Forests, Support Vector Machines, K-Means Clustering sowie
> neuronale Netze und Deep Learning

Weiterhin:

> Ein Verständnis der Modellevaluation, einschließlich Cross-Validation, relevanter Metriken (Accuracy,
> Precision, Recall, F1-Score) und des Bias-Variance Tradeoffs

Genau das soll doch gelehrt werden im LZ 3-11. Wieso benötige ich vorab schon ein Verständnis davon? Allerhöchstens würde ich hier Überblickswissen verlangen.

Weiterhin:

> Erfahrung mit typischen Frameworks wie scikit-learn, TensorFlow und PyTorch

Das wäre mir nicht spezifisch genug. Erfahrung kann hier alles sein von einmal ausprobiert bis hin zu jahrelanger Erfahrung. Hier würde ich ebenfalls, maximal, Überblickswissen einfordern.

Weiterhin:

> Vertrautheit mit Jupyter Notebooks oder ähnlichen Werkzeugen zur schnellen
Anwendungsentwicklung (Rapid Application Development, RAD)

Auch hier zu unspezifisch IMHO. Überblickswissen genügt hier vollauf, finde ich.

Weiterhin:

> Ein grundlegendes Verständnis der technischen Verfahren hinter Künstlicher Intelligenz, einschließlich
> Maschinellen Lernens (insbesondere Deep Learning und LLMs) sowie Ansätzen im Bereich Natural
> Language Processing (NLP) und Transformationsmodellen

Finde ich gut, würde hier aber auch aus Gründen der Konsistenz "Überblickswissen" verlangen. Je weniger verschiedene Begriff für Wissensstände wir nutzen, desto einfacher einzuordnen für Teilnehmer:innen und Trainer:innen.

> Transformationsmodelle

Geht es hier um Transformermodelle?

Weiterhin:

> Praktische Erfahrung im Training von Künstlichen Intelligenz-Modellen

Das finde ich definitiv viel zu hart. Ich glaube, vom adressierbaren Markt dieses Trainings werden das allerhöchstens 1% der Teilnehmer:innen vorweisen können.

**Comment by mattes3** [comment]
Hallo @youngbrioche , vielen Dank, Du sprichst mir aus der Seele. Wenn ich als TN die jetzigen Voraussetzungen läse, würde ich fast sofort vom Training absehen.

Ich finde, Architekten müssen nicht selbst ML-Experten sein. Ich kann heute recht genau erkennen, was z.B. ein LLM leisten kann und was nicht, obwohl ich die Transformer-Architektur nicht zu 100% verstanden habe. Dafür muss ich sie nur genügend oft angewendet oder in Anwendung gesehen haben.

Vergleich mit Compilern: Ich muss nicht genau verstehen, wie ein Compiler intern funktioniert, selbst wenn ich ihn den ganzen Tag verwende. Als Architekt muss ich nur einschätzen können, was damit machbar ist, wo die Grenzen und Risiken liegen, und wie ich die zu planende Lösung strukturieren könnte.

**Comment by mahboubagharbi** [comment]
Wir werden diese Anmerkungen im nächsten Release prüfen, das möglicherweise in 2-3 Monate veröffentlicht wird. Viele Teilnehmer lesen den Lehrplan nicht einmal. ;-)

**Comment by programming-wolf** [comment]
Erledigt mit https://github.com/isaqb-org/curriculum-swarc4ai/pull/30
